packages/plan-form/src/Form/tests/index.test.tsx

```diff
@@ -0,0 +1,614 @@
+    ).toMatchSnapshot('name error');
+
+    // title is required
+    expect(
+      (wrapper.find('FormItem[name="title"] FormItemInput').props() as any).errors
+    ).toMatchSnapshot('title error');
+
+    // description is required
+    expect(
+      (wrapper.find('FormItem[name="description"] FormItemInput').props() as any).errors
+    ).toMatchSnapshot('title error');
+
+    // let us cause errors for other required fields and ascertain that they are indeed validated
+
+    // Remove the date field value
+    wrapper.find('#date input').simulate('change', { target: { name: 'date', value: '' } });
+
+    // Remove the dateRange field value
+    formInstance.setFieldsValue({ dateRange: '' });
+
+    // Remove the interventionType field value
+    formInstance.setFieldsValue({ interventionType: '' });
+    wrapper.find('form').simulate('submit');
+
+    await act(async () => {
+      await new Promise<any>((resolve) => setImmediate(resolve));
+      wrapper.update();
+    });
+
+    formInstance = (wrapper.find(Form).props() as any).form;
+    errors = formInstance.getFieldsError();
+
+    fieldsWithErrors = errors.filter((error) => error.errors.length > 0);
+
+    // we now have some more errors
+    expect(fieldsWithErrors).toMatchSnapshot('more Errors');
+
+    // date is required
+    expect(
+      (wrapper.find('FormItem[name="date"] FormItemInput').props() as any).errors
+    ).toMatchSnapshot('date error');
+
+    // interventionType is required
+    expect(
+      (wrapper.find('FormItem[name="interventionType"] FormItemInput').props() as any).errors
+    ).toMatchSnapshot('interventionType error');
+
+    // dateRange is required
+    expect(
+      (wrapper.find('FormItem[name="dateRange"] FormItemInput').props() as any).errors
+    ).toMatchSnapshot('dateRange required error');
+
+    // description is required
+    expect(
+      (wrapper.find('FormItem[name="description"] FormItemInput').props() as any).errors
+    ).toMatchSnapshot('description required error');
+
+    // next we set wrong values for fields that expect specific values
+
+    // Set wrong interventionType field value
+    formInstance.setFieldsValue({
+      interventionType: 'Oov',
+    });
+    wrapper.find('form').simulate('submit');
+
+    await act(async () => {
+      await new Promise<any>((resolve) => setImmediate(resolve));
+      wrapper.update();
+    });
+
+    formInstance = (wrapper.find(Form).props() as any).form;
+    errors = formInstance.getFieldsError();
+
+    fieldsWithErrors = errors.filter((error) => error.errors.length > 0);
+
+    // we now have some more errors
+    expect(fieldsWithErrors).toMatchSnapshot('invalid data type Errors');
+  });
+
+  it('Auto-setting name and title field values works', async () => {
+    const wrapper = mount(
+      <MemoryRouter>
+        <PlanForm />
+      </MemoryRouter>
+    );
+    // Set title for the plan
+    wrapper
```

Comment:
Should we note in an issue or elsewhere that this does not work for Reveal-style plans where the name and title generated potentially include the jurisdiction name and FI status?